functions/check-started/index.js

```diff
@@ -1,11 +1,46 @@
 'use strict'
 
-const uuid = require('uuid/v4')
+const azureStorage = require('azure-storage')
+const bluebird = require('bluebird')
 const moment = require('moment')
+const sqlService = require('less-tedious')
+const uuid = require('uuid/v4')
+const winston = require('winston')
+const { TYPES } = require('tedious')
+
+winston.level = 'error'
+const config = require('../config')
+sqlService.initialise(config)
+
+const checkStatusTable = '[checkStatus]'
+const checkTable = '[checkFormAllocation]'
+const schema = ['mtc_admin']
+const preparedCheckTable = 'preparedCheck'
+let azureTableService
+initAzureTableService()
 
-module.exports = function (context, checkStartMessage) {
+module.exports = async function (context, checkStartMessage) {
   context.log('check-started message received', checkStartMessage.checkCode)
-  // TODO: purpose: process check start messages and put into pupilEvents table
+
+  // Update the admin database to update the check status to Check Started
+  try {
+    await updateAdminDatabaseForCheckStarted(checkStartMessage.checkCode, context.log)
+    context.log('SUCCESS: Admin DB updated')
+  } catch (error) {
+    context.log.error(`ERROR: unable to update admin db for [${checkStartMessage.checkCode}]`)
+    throw error
+  }
+
+  // Delete the row in the preparedCheck table - prevent pupils logging in again.
+  try {
+    await deleteFromPreparedCheckTableStorage(checkStartMessage.checkCode, context.log)
+    context.log('SUCCESS: Table storage cache deleted from preparedCheck table')
+  } catch (error) {
+    context.log.error(`ERROR: unable to delete from table storage for [${checkStartMessage.checkCode}]`)
+    throw error
+  }
+
+  // Store the raw message to an audit log
   context.bindings.pupilEventsTable = []
   const entity = {
     PartitionKey: checkStartMessage.checkCode,

```

Comment:
you will need rowKey also i believe, as that is the unique identifier of the row in the partition.  you can just add a UUID, as checkCode is good for partition.

Reply:
This code is unchanged, so not in the diff, and it includes the RowKey 😄 :

```
const entity = {
    PartitionKey: checkStartMessage.checkCode,
    RowKey: uuid(),
    eventType: 'check-started',
    payload: JSON.stringify(checkStartMessage),
    processedAt: moment().toDate()
  }
```

Reply:
ahhh!